suite-native/module-authorize-device/src/screens/connect/ConnectAndUnlockDeviceScreen.tsx

```diff
@@ -50,6 +53,15 @@ export const ConnectAndUnlockDeviceScreen = () => {
         }
     }, [isDeviceAuthorized, device, dispatch, isFocused]);
 
+    useEffect(() => {
+        if (isFocused && isDeviceConnected) {
+            // When selected device become connected, we need to navigate out of this screen.
+            if (navigation.canGoBack()) {
+                navigation.goBack();
```

Comment:
Wont this make previous useEffect irrelevant?  

``` if (isFocused && device && !isDeviceAuthorized) {```

Reply:
Yes! Thanks for the catch 👍

Reply:
7b95a242ce99463f8145434b837f96ecfe336838